src/public/modules/forms/admin/controllers/view-responses.client.controller.js

```diff
@@ -47,6 +47,32 @@ function ViewResponsesController(
 
   vm.datePicker = { date: { startDate: null, endDate: null } }
 
+  if (typeof BroadcastChannel === 'function') {
+    vm.privateKeyChannel = new BroadcastChannel('formsg_private_key_sharing')
+    vm.privateKeyChannel.onmessage = function (e) {
+      if (e.data.action === 'broadcastKey') {
+        if (vm.encryptionKey === null && vm.myform._id === e.data.formId) {
```

Comment:
thinking through this bit of logic again - the user needs to already have the 2nd tab open prior to entering the key? so broadcast channel on its own wouldn't be able to help with opening a link from the CSV, right?

Reply:
No, the user doesn’t. When the 2nd tab loads, it will send a `requestKey` message and any already opened tab should respond with another `broadcastKey` message.